sanitize_html/lib/src/sane_html_validator.dart

```diff
@@ -217,35 +220,53 @@ class SaneHtmlValidator implements html.NodeValidator {
     @required this.allowClassName,
   });
 
-  @override
-  bool allowsAttribute(html.Element element, String attribute, String value) {
-    if (!_allowedElements.contains(element.tagName)) {
-      return false;
-    }
-    if (_alwaysAllowedAttributes.contains(attribute)) {
-      return true;
-    }
-    // Special validators for id and class on all elements
-    if (attribute == 'id') {
-      return allowElementId(element.id);
+  String sanitize(String htmlString) {
+    final root = html_parser.parseFragment(htmlString);
+    _sanitize(root);
```

Comment:
Please `assert(root is Element)`, if this is not the case we're in big trouble.

Maybe if you start with `<html></html>` it won't be? or what about `<-- bla bla--->`... We'll probably need a few test cases for this as top-level.

Reply:
What if the htmlString is `<span></span><span></span>` won't the `root` be a `DocumentFragment`?

Reply:
`parseFragment()` returns with a `DocumentFragment`, not `Element`, and `<span></span><span></span>` is already handled by that. I've added a test case to handle multiple top-level elements.